lib/zendesk_api/collection.rb

```diff
@@ -356,30 +356,27 @@ def intentional_obp_request?
       Helpers.present?(@options["page"]) && !cbp_request?
     end
 
-    def should_try_cbp?(path_query_link)
-      not_supported_endpoints = %w[show_many]
-      not_supported_endpoints.none? { |endpoint| path_query_link.end_with?(endpoint) }
+    def supports_cursor_based_pagination?
+      @resource_class.const_defined?(:CBP_ACTIONS) && @resource_class.const_get(:CBP_ACTIONS).any? { |supported_path| path.end_with?(supported_path) }
+    end
+
+    def first_of_multiple_pages?
```

Comment:
Not bad.

* Is this for the first page only?
* Do we know there are multiple pages?

Can you add a method comment to explain why `@next_page.nil?` defines the first of multiple pages?